src/wireviz/wv_html.py

```diff
@@ -21,14 +21,19 @@ def generate_html_output(
     bom_list: List[List[str]],
     metadata: Metadata,
     options: Options,
+    source: Union[str, Path] = None,
 ):
     # load HTML template
     templatename = metadata.get("template", {}).get("name")
+    template_search_paths = [ Path(filename).parent, Path(__file__).parent / "templates"]
+
+    if source is not None:
+        template_search_paths.insert(0, Path(source).parent)
+
     if templatename:
         # if relative path to template was provided, check directory of YAML file first, fall back to built-in template directory
         templatefile = smart_file_resolve(
-            f"{templatename}.html",
-            [Path(filename).parent, Path(__file__).parent / "templates"],
+            f"{templatename}.html", template_search_paths
```

Comment:
Remember inserting `from wireviz.wv_bom import make_list` in the import section above and run `isort *.py` to format and sort them.
```suggestion
            f"{templatename}.html",
            make_list(input_folders) + [Path(__file__).parent / "templates"],
```
My suggestion here does not add the output folder in the list when different because I don't see any use case where that might be useful, but please argue against my suggestion.